controlplane/kubeadm/internal/cluster.go

```diff
@@ -70,6 +71,7 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 	// TODO(chuckha): Inject this dependency.
 	// TODO(chuckha): memoize this function. The workload client only exists as long as a reconciliation loop.
 	restConfig, err := remote.RESTConfig(ctx, m.Client, clusterKey)
+	restConfig.Timeout = 30 * time.Second
```

Comment:
Note that this timeout seems to be only used for etcd client generator, the `remote.NewClusterClient` still has no timeout set

cc @yastij @randomvariable